cfme/common/vm.py

```diff
@@ -67,6 +67,90 @@ def all_types(template=False):
     return all_types
 
 
+def set_retirement_date(obj, when=None, offset=None, warn=None):
```

Comment:
REQUIRED: please turn these functions back into vm/instance methods.

Reply:
I moved these methods so that they could be called both for a single VM/Instance plenty and for a list of one or more VMs/Instances.

Reply:
@tpapaioa noted on your intended use, but this moves us toward a functional implementation instead of an OO one.

These methods, ideally, would operate on the collection or entity for a VM, so that a filtered collection or a single vm entity could have the same `retire/set_retirement_date` method called on the instance, instead of the instance being passed.

The common functionality (param checking, form interaction) could be defined as module methods, but would need to be wrapped into instance methods for the collection/entity.

This would require better filtering support on the collections - right now I'm not aware of a way to filter the vms collections based on partially matched names.

Reply:
If we're going to accept/allow this implementation of these, I'd like to see:

1. docblock
2. Better name for the first arg, since it could be an entity or list of entities.

Reply:
if you would like to do that for a list of VMs. you can add collection method and pass it a list.

Reply:
Added retire and set_retirement_date (instance) methods (with docblocks) to the VMCollection class, which InstanceCollection and InfraVMCollection inherit.